tests/backends/aiida_django/migrations/test_migrations_0046_inline_calc_attributes.py

```diff
@@ -0,0 +1,98 @@
+from .test_migrations_common import TestMigrations
+
+
+class TestCalcFunctionAttributesMigration(TestMigrations):
+    """Test the migration for the attributes of InlineFunctions that have been transformed into CalcFunctions."""
+
+    migrate_from = '0045_dbgroup_extras'
+    migrate_to = '0046_inline_calc_attributes'
+
+    @staticmethod
+    def get_source_code(full_file=False):
+        """Returns a string containing the source code for the calcfunction"""
+
+        # yapf: disable
+        text_i = ('from aiida import orm\n'
+                  'from aiida.engine import calcfunction\n'
+        )
+        text_t = ('@calcfunction\n'
+                  'def sum_inline( int1, int2 ):\n'
+                  '    return int1 + int2\n'
+        )
+        text_f = ('num1 = orm.Int(2)\n'
+                  'num2 = orm.Int(3)\n'
+                  'num3 = sum_inline( int1, int2 )\n'
+                  'print(num3)\n'
+        )
+        # yapf: enable
+
+        if full_file:
+            text_t = text_i + '\n' + text_t + '\n' + text_f
+
+        return text_t
+
+    def setUpBeforeMigration(self):
+        calcnode = self.DbNode(
+            node_type='process.calculation.calcfunction.CalcFunctionNode.',
+            user_id=self.default_user.id,
+            attributes={
+                'sealed': True,
+                'namespace': '__main__',
+                'source_code': self.get_source_code(full_file=False),
+                'source_file': self.get_source_code(full_file=True),
+                'function_name': 'sum_inline',
+                'first_line_source_code': 4,
+            },
+        )
+        calcnode.save()
+        self.calcnode_id = calcnode.id
+        #print(self.load_node(self.calcnode_id).attributes)
+        #calcnode.attributes['function_namespace'] = calcnode.attributes['namespace']
+        #calcnode.save()
+        #print(self.load_node(self.calcnode_id).attributes)
+
+    def test_data_migrated(self):
+        """Test that the model now has an extras column with empty dictionary as default."""
+        calcnode = self.load_node(self.calcnode_id)
+
+        # Pre existing and renamed attributes
+        self.assertEqual(calcnode.attributes['sealed'], True)
+        self.assertEqual(calcnode.attributes['function_name'], 'sum_inline')
+        self.assertEqual(calcnode.attributes['function_namespace'], '__main__')
+        self.assertEqual(calcnode.attributes['function_starting_line_number'], 4)
+
+        # Newly determined attributes
+        self.assertEqual(calcnode.attributes['exit_status'], 0)
+        self.assertEqual(calcnode.attributes['process_state'], 'finished')  #USE ENUM?
+        self.assertEqual(calcnode.attributes['process_label'], 'Legacy InlineCalculation')
+        self.assertEqual(calcnode.attributes['function_number_of_lines'], 3)
+
+        # Old attributes that got renamed or removed
+        with self.assertRaises(KeyError):
+            _ = calcnode.attributes['namespace']
```

Comment:
Would be more natural and shorter to write this as 
```suggestion
        assert 'namespace' not in calcnode.attributes
```

Reply:
Mmm yes you are right, but my only question is that `assert 'namespace' not in calcnode.attributes` looks like pytest syntax, can I use that inside an `AiidaTestCase` subclass? (Or even if I could, should I be mixing the two syntaxes?)

Reply:
there is nothing specifically pytest about using `assert`. The `assert` is a normal Python construct. `pytest` just catches the `AssertionError` that get's raised and deals with it. I think you should be able to use it just fine in unittest based test.

Reply:
Should I also then change all the `self.assertEqual(a,b)` for simpler `assert a == b`?

Reply:
fair enough, you could go for `self.assertFalse('namespace' in calcnode.attributes')` this way you still use the same syntax but it is just a single line per test. Bonus points if you put it in a simple loop for all keys that should be absent ;)

Reply:
Haha I was actually honestly asking to make the change like a "partial migration to pytest", was not trying to argue against it =P. But ok, I'll do the `self.assertFalse`.

> Bonus points if you put it in a simple loop for all keys that should be absent ;)

Yes brownie points gimme :grimacing: